websubhub-ballerina/commons.bal

```diff
@@ -281,8 +305,24 @@ public final InternalUnsubscriptionError INTERNAL_UNSUBSCRIPTION_ERROR = error I
 public final UnsubscriptionDeniedError UNSUBSCRIPTION_DENIED_ERROR = error UnsubscriptionDeniedError("Unsubscription denied");
 
 # Record to represent client configuration for HubClient / PublisherClient
+# 
+# + httpVersion - The HTTP version understood by the client
+# + http1Settings - Configurations related to HTTP/1.x protocol
+# + http2Settings - Configurations related to HTTP/2 protocol
+# + timeout - The maximum time to wait (in seconds) for a response before closing the connection
+# + poolConfig - Configurations associated with request pooling
+# + retryConfig - Configurations associated with retrying
+# + responseLimits - Configurations associated with inbound response size limits
+# + secureSocket - SSL/TLS related options
 public type ClientConfiguration record {|
-    *http:ClientConfiguration;
+    string httpVersion = HTTP_1_1;
+    http:ClientHttp1Settings http1Settings = {};
+    http:ClientHttp2Settings http2Settings = {};
+    decimal timeout = 60;
+    http:PoolConfiguration? poolConfig = ();
+    http:RetryConfig? retryConfig = ();
+    http:ResponseLimitConfigs responseLimits = {};
+    http:ClientSecureSocket? secureSocket = ();
```

Comment:
IMO, it would be better to apply the following change (`optional` vs `nilable`) wherever applicable. This won't be affect for the user, but the module owner may have to update the implementation logic. Ex. with previous config we know the field does exist always and it can be `()` or the expected value. But now the field may exist or not. So, accessing field like `config.retryConfig` can give an error because the field does not exist. @shafreenAnfar @ayeshLK WDYT? 

Applicable for all the modules though. If the change is huge, it is okay to address this in a separate PR also.
```suggestion
    string httpVersion = HTTP_1_1;
    http:ClientHttp1Settings http1Settings = {};
    http:ClientHttp2Settings http2Settings = {};
    decimal timeout = 60;
    http:PoolConfiguration poolConfig?;
    http:RetryConfig retryConfig?;
    http:ResponseLimitConfigs responseLimits = {};
    http:ClientSecureSocket secureSocket?;
```

Reply:
Yeah this is the idiomatic approach. Will change the codebase accordingly.